dataland-frontend/src/components/forms/parts/kpiSelection/DataPointFormField.vue

```diff
@@ -0,0 +1,154 @@
+                validation-label="Page"
+                step="1"
+                min="0"
+                validation="min:0"
+              />
+            </div>
+          </div>
+        </FormKit>
+      </div>
+
+      <!-- Data quality -->
+      <div class="mb-4" data-test="dataQuality">
+        <UploadFormHeader
+          :label="`${label} Quality`"
+          description="The level of confidence associated to the value."
+          :is-required="isDataQualityRequired"
+        />
+        <div class="md:col-6 col-12 p-0">
+          <FormKit
+            type="select"
+            :modelValue="!isDataQualityRequired ? 'NA' : currentQualityValue"
+            name="quality"
+            :validation="isDataQualityRequired ? 'required' : ''"
+            validation-label="Data quality"
+            placeholder="Data quality"
+            :options="qualityOptions"
+          />
+        </div>
+      </div>
+      <div class="form-field">
+        <FormKit
+          type="textarea"
+          name="comment"
+          placeholder="(Optional) Add comment that might help Quality Assurance to approve the datapoint. "
+        />
+      </div>
+    </FormKit>
+  </template>
+
+  <template v-else>
+    <FormKit type="group" :name="name">
+      <UploadFormHeader :label="label" :description="description ?? ''" :is-required="required" />
+      <FormKit
+        type="text"
+        :name="name"
+        v-model="currentValue"
+        :validation-label="validationLabel ?? label"
+        :validation="`number|${validation}`"
+        :placeholder="placeholder"
+        :inner-class="innerClass"
+      />
+    </FormKit>
+  </template>
+</template>
+
+<script lang="ts">
+import { defineComponent } from "vue";
+import UploadFormHeader from "@/components/forms/parts/elements/basic/UploadFormHeader.vue";
+import { FormKit } from "@formkit/vue";
+import { QualityOptions } from "@clients/backend";
+import { selectNothingIfNotExistsFormKitPlugin } from "@/utils/FormKitPlugins";
+import { DataPointFormFieldProps } from "@/components/forms/parts/fields/FormFieldProps";
+
+export default defineComponent({
+  name: "DataPointFormField",
+  components: { UploadFormHeader, FormKit },
+  inject: {
+    injectReportsName: {
+      from: "namesOfAllCompanyReportsForTheDataset",
+      default: [] as string[],
+    },
+  },
+  computed: {
+    displayreportsName(): string[] {
+      return (this.reportsName ?? this.injectReportsName) as string[];
+    },
+    isDataQualityRequired(): boolean {
+      return this.currentValue !== "";
+    },
+  },
+  data() {
+    return {
+      qualityOptions: Object.values(QualityOptions).map((qualityOption: string) => ({
+        label: qualityOption,
+        value: qualityOption,
+      })),
+      currentValue: "",
+      currentUnit: "",
+      currentReportValue: "",
+      currentPageValue: "",
+      currentQualityValue: "",
+    };
+  },
+  emits: ["reportsUpdated"],
+  props: {
+    ...DataPointFormFieldProps,
```

Comment:
These props can be put here directly, because they are only used for this component here.

Reply:
Don't we have a strategy to import props from `FormFieldProps.ts`? At first we had the props here,but remembered the strategy and decided to move it to the file with the rest of the form props.

Reply:
No as far as I know this is not the strategy in general.
I think the `FormFieldProps.ts` has been introduced to avoid setting the same props again and again in some cases.
Some prop-combinations are repeated in several components. Then it makes sense to centrally maintain them in the `FormFieldProps.ts`.

But in this case, it is not repeated.

I think Marcin has already moved them, so I can close this conversation.